geonode/geoserver/manager.py

```diff
@@ -264,3 +266,26 @@ def set_time_info(self, method: str, uuid: str, /, instance: ResourceBase = None
                 logger.exception(e)
                 return None
         return instance
+
+    def revise_resource_value(self, _resource, files, user, action_type):
```

Comment:
In the case of a "replace" action, we should update the "ResourceBase.FILES" also.